stdlib/http/src/main/java/org/ballerinalang/net/http/websocket/WebSocketUtil.java

```diff
@@ -221,7 +251,462 @@ public static ErrorValue createErrorCause(String message, String reason, BPackag
+        retryConnectorConfig.setMaxInterval(maxInterval);
+    }
+
+    /**
+     * Set value of the maxAttempts in the retry config.
+     *
+     * @param maxAttempts a maximum number of retry attempts
+     * @param retryConnectorConfig retry connector config
+     */
+    private static void setMaxAttempts(int maxAttempts, RetryContext retryConnectorConfig) {
+        if (maxAttempts < 0) {
+            logger.warn("{} The maxAttempts[{}] value is set to 0", STATEMENT_FOR_MAX_ATTEPTS,
+                    maxAttempts);
+            maxAttempts = 0;
+        }
+        retryConnectorConfig.setMaxAttempts(maxAttempts);
+    }
+
+    /**
+     * Populate the failover config.
+     *
+     * @param clientEndpointConfig a client endpoint config
+     * @param failoverClientConnectorConfig a failover client connector config
+     * @param targetUrls target urls
+     */
+    public static void populateFailoverConnectorConfig(MapValue<String, Object> clientEndpointConfig,
+                                                       FailoverContext failoverClientConnectorConfig,
+                                                       ArrayList<String> targetUrls) {
+        int failoverInterval = Integer.parseInt(clientEndpointConfig.get(WebSocketConstants.FAILOVER_INTEVAL)
+                .toString());
+        if (failoverInterval < 0) {
+            logger.warn("{} The failoverInterval [{}] value is set to 1.0",
+                    STATEMENT_FOR_FAILOVER_INTERVAL, failoverInterval);
+            failoverInterval = 1000;
+        }
+        failoverClientConnectorConfig.setFailoverInterval(failoverInterval);
+        failoverClientConnectorConfig.setTargetUrls(targetUrls);
+    }
+
+    /**
+     * Validate and create the webSocket service.
+     *
+     * @param clientEndpointConfig a client endpoint config
+     * @param strand a strand
+     * @return webSocketService
+     */
+    public static WebSocketService validateAndCreateWebSocketService(MapValue<String, Object> clientEndpointConfig,
+                                                                     Strand strand) {
+        Object clientService = clientEndpointConfig.get(WebSocketConstants.CLIENT_SERVICE_CONFIG);
+        if (clientService != null) {
+            BType param = ((ObjectValue) clientService).getType().getAttachedFunctions()[0].getParameterType()[0];
+            if (param == null || !(WebSocketConstants.WEBSOCKET_CLIENT_NAME.equals(
+                    param.toString()) || WebSocketConstants.WEBSOCKET_FAILOVER_CLIENT_NAME.equals(
+                    param.toString()))) {
+                throw new WebSocketException("The callback service should be a WebSocket Client Service");
+            }
+            return new WebSocketService((ObjectValue) clientService, strand.scheduler);
+        } else {
+            return new WebSocketService(strand.scheduler);
+        }
+    }
+
+    public static WebSocketClientConnector getWebSocketClientConnector(String remoteUrl,
```

Comment:
Let's rename to make the intent clear
```suggestion
    public static WebSocketClientConnector createWebSocketClientConnector(String remoteUrl,
```